src/sql/parts/disasterRecovery/restore/common/restoreServiceImpl.ts

```diff
@@ -153,9 +160,29 @@ export class RestoreDialogController implements IRestoreDialogController {
 			restoreOption.taskExecutionMode = TaskExecutionMode.executeAndScript;
 		}
 
-		this._restoreService.restore(this._ownerUri, restoreOption);
-		let restoreDialog = this._restoreDialogs[this._currentProvider];
-		restoreDialog.close();
+		this._restoreService.restore(this._ownerUri, restoreOption).then(() => {
+			const self = this;
+			let connectionProfile = self._connectionService.getConnectionProfile(self._ownerUri);
+			let activeNode = self._objectExplorerService.getObjectExplorerNode(connectionProfile);
+			this._taskService.onTaskComplete(response => {
+				if (this.isSuccessfulRestore(response) && activeNode) {
+					self._objectExplorerService.refreshTreeNode(activeNode.getSession(), activeNode).then(result => {
+						self._objectExplorerService.getServerTreeView().refreshTree();
```

Comment:
Why do we need to call `refreshTree` here? Does the call to `refreshTreeNode` just update the backing model but not actually show the changes in the view?

It seems unlikely that anyone would ever want to call `refreshTreeNode` on the object explorer service and not have the refresh actually show up in the view, so maybe it's best to just add this call to the `refreshTreeNode` code inside of OE service?

Reply:
No refreshTreeNode doesn't update the view. I've checked the OE code as well, the ITree has to be refreshed for the updated view.

Reply:
Is refreshTree() trying to refresh the whole tree?

---
In reply to: [166411781](https://github.com/Microsoft/sqlopsstudio/pull/607#discussion_r166411781) [](ancestors = 166411781)